packages/web-components/fast-foundation/src/tree-view/tree-view.ts

```diff
@@ -233,6 +233,9 @@ export class TreeView extends FoundationElement {
 
     private handleItemSelected = (e: CustomEvent): void => {
         const newSelection: TreeItem = e.target as TreeItem;
+        if (newSelection.disabled) {
+            e.stopPropagation();
+        }
```

Comment:
Can you share the reasoning for handling this programmatically vs. using `pointer-events: none` elsewhere? This seems reasonable, but it would be good to ensure we have a consistent approach to handling these so there is an intuitive developer experience for library authors and intuitive UX for web authors.

Reply:
Sure, the reason `pointer-events: none` is a bad approach here is because of nested Tree Items. If a disabled tree is nested the `pointer-events: none` successfully prevents the vent from firing on that item but the event fires on the parent tree item since the parent tree item hit target is the size of itself and all its children.

<img width="728" alt="Screen Shot 2021-11-29 at 11 54 06 AM" src="https://user-images.githubusercontent.com/37851214/144094284-74c34b1c-59cc-4157-aa28-4266f7607fca.png">

Reply:
Another option would be to move the click event on treeItem from the root template to the `positioning-region` This would ensure that only clicks happen on the item it self and aren't passed from a tree items nested items

Reply:
I'm thinking the tree items shouldn't select based on clicks in the nested child region.  If someone introduced styling that added margins between child items you wouldn't expect clicks between two items to select the parent, for example.